wrapper/src/test/java/software/amazon/jdbc/util/ConnectionUrlParserTest.java

```diff
@@ -90,4 +100,15 @@ private static Stream<Arguments> passwordUrls() {
         Arguments.of("protocol//url/db?pass=foo", null)
     );
   }
+
+  private static Stream<Arguments> encodedParams() {
+    return Stream.of(
+        Arguments.of("protocol//host/db?param=" + StringUtils.encode("value$"), "value$"),
+        Arguments.of("protocol//host/db?param=" + StringUtils.encode("value_"), "value_"),
+        Arguments.of("protocol//host/db?param=" + StringUtils.encode("value?"), "value?"),
+        Arguments.of("protocol//host/db?param=" + StringUtils.encode("value&"), "value&"),
+        Arguments.of("protocol//host/db?param=" + StringUtils.encode("value"
+        + new String(Character.toChars(0x1f604))), "value" + new String(Character.toChars(0x1f604)))
```

Comment:
nit: I'd replace "new String(Character.toChars(0x1f604)))" with actual string representing this Unicode char. It's just for readability of the test.